squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/mime/FiniteStateMachineFactory.java

```diff
@@ -0,0 +1,145 @@
+package org.dice_research.squirrel.analyzer.mime;
+
+import org.apache.jena.riot.RDFLanguages;
+
+import java.util.ArrayList;
+
+/**
+ * Building different finite state machines.
+ *
+ * @author Abhishek Hassan Chandrashekar (abhihc@mail.uni-paderborn.de).
+ */
+public class FiniteStateMachineFactory {
+
+    static public FiniteStateMachine create(String mimeType) {
+        return new FiniteStateMachineFactory().buildStateMachine(mimeType);
+    }
+
+    private FiniteStateMachine buildStateMachine(String mimeType) {
+        switch (mimeType) {
+            case "RDF/XML":
+                return buildRDFXMLStateMachine();
+            case "Turtle":
+                return buildTurtleStateMachine();
+            case "N-Triples":
+                return buildNTriplesStateMachine();
+            case "RDF/JSON":
+                return buildRDFJSONStateMachine();
+            case "JSON-LD":
+                return buildJSONLDStateMachine();
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Builds a finite state machine to validate a simple
+     * RDFXML file
+     */
+    private FiniteStateMachine buildRDFXMLStateMachine() {
+
+        ArrayList<State> listOfStates = new ArrayList<>();
+
+        String[] validRules = {"<", "\\?", "x", "m", "l"};
+        String[] invalidRules = {"[^<]", "[^\\?]", "[^x]", "[^m]", "[^l]"};
+
+        populateStates(listOfStates, validRules.length);
+        populateTransitions(listOfStates, validRules, invalidRules);
+
+        return new Automata(listOfStates.get(0), RDFLanguages.RDFXML);
+    }
+
+    /**
+     * Builds a finite state machine to validate a simple
+     * NTRIPLES file
+     */
+
+    private FiniteStateMachine buildNTriplesStateMachine() {
```

Comment:
Ok, so the Automata's written are completely focused on triple which has the subject, object, and predicate in URI format. I focused completely on the resources which were available in the project.

Reply:
But where is the sense in stating that we only crawl resources that we already have in our project? :confused: 

I asked you to have a look at the specifications of the languages and after we discussed the first ideas that you had regarding this task, I asked you again to take a look at them, e.g., to make sure that comment lines do not create issues, etc. We also discussed that your first test cases where not sufficient since you had only one simple example for each language and I asked you to add more test cases to make sure that you cover the complete language.

However, since today is the deadline for the code, I do not see a possibility to fix these issues in the remaining time :disappointed:

Reply:
I was already wondering, how this "magic" works :smile: 

Unfortunately, adding this to the MimeTypeDetector class has two disadvantages.
1. You are splitting up the definition of the languages. One part of the languages is now part of the automatons while another part is in the MimeTypeDetector class.
2. You are defining lines starting with `#` as general comment lines. However, this is not true for all the languages, e.g., XML. You could easily imagine a situation in which this can lead to a large amount of issues.

Reply:
Yes, I get it. I overlooked it in a generic manner.